sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystemProvider.java

```diff
@@ -178,13 +162,17 @@
 
     private final ConcurrentMap<String, FileSystem> openFileSystems;
 
+    private static Map<String, Object> defaultConfigurations = null;
+
 
     // Specs require a public zero argument constructor.
     /**
      * Creates an AzureFileSystemProvider.
      */
     public AzureFileSystemProvider() {
         this.openFileSystems = new ConcurrentHashMap<>();
+        Configuration config = Configuration.getGlobalConfiguration();
```

Comment:
Are there scenarios where user can call `AzureFileSystemProvider` ctor directly? If yes we could consider adding `AzureFileSystemProvider(Configuration)` overload in case somebody would like to customize configuration source.

Btw. what do we do with `config` ? it doesn't seem to be consumed anyhow.

Reply:
This was left over from when the original request was to set defaults from the environment, but the customer ask has changed to setting them on a static method. I thought I had reverted all the commits about environment reading but it looks like I missed one. 

But in general no, there customer does not generally use the constructor directly. The jvm's ClassLoader will find it an initialize it